File: code/scripts/sync-description.js
import fs from 'fs'
import path from 'path'

const authorPath = path.join(__dirname, '../data/authors/default.mdx')
const siteMetadataPath = path.join(__dirname, '../data/siteMetadata.js')

// Read author file
const file = fs.readFileSync(authorPath, 'utf8')
// Remove frontmatter
const content = file.replace(/---[\s\S]*?---/, '').trim()
// Get first non-empty paragraph
const paragraphs = content
  .split(/\n\n+/)
  .map((p) => p.trim())
  .filter(Boolean)
const firstParagraph = paragraphs.length > 0 ? paragraphs[0] : ''

// Read siteMetadata.js
let siteMetadataSrc = fs.readFileSync(siteMetadataPath, 'utf8')

// Replace the description field (assumes it is a single- or multi-line string)
siteMetadataSrc = siteMetadataSrc.replace(
  /description:\s*([`'"])([\s\S]*?)(\1),/,
  `description: '${firstParagraph.replace(/'/g, "\\'")}',`
)

fs.writeFileSync(siteMetadataPath, siteMetadataSrc)
console.log('siteMetadata.js description updated from default.mdx')

File: code/scripts/utils/sync_description.py
#!/usr/bin/env python3
import json
import os
import re

ROOT = os.getcwd()
author_path = os.path.join(ROOT, "data", "non-public", "authors", "default.mdx")
site_metadata_path = os.path.join(ROOT, "data", "non-public", "siteMetadata.json")

with open(author_path) as f:
    content = f.read().replace("\r\n", "\n")

body = re.sub(r"---[\s\S]*?---", "", content).strip()
paragraphs = [p.strip() for p in re.split(r"\n\n+", body) if p.strip()]
first_paragraph = paragraphs[0] if paragraphs else ""

with open(site_metadata_path) as f:
    metadata = json.load(f)

metadata["description"] = first_paragraph

with open(site_metadata_path, "w") as f:
    json.dump(metadata, f, indent=2)
    f.write("\n")

print("siteMetadata.json description updated from default.mdx")
